Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/InteropCodeGenerator/SpatialGDKEditorInteropCodeGenerator.cpp

```diff
@@ -127,36 +119,110 @@ FString GenerateIntermediateDirectory()
 	return AbsoluteCombinedIntermediatePath;
 }
 
-bool SpatialGDKGenerateInteropCode()
+void CreateSchemaDatabase(TArray<UClass*> Classes)
 {
-	const USpatialGDKEditorToolbarSettings* SpatialGDKToolbarSettings = GetDefault<USpatialGDKEditorToolbarSettings>();
-	TArray<UClass*> InteropGeneratedClasses;
+	AsyncTask(ENamedThreads::GameThread, [Classes]{
+		FString PackagePath = TEXT("/Game/Spatial/SchemaDatabase");
+
+		UPackage *Package = CreatePackage(nullptr, *PackagePath);
+
+		USchemaDatabase* SchemaDatabase = NewObject<USchemaDatabase>(Package, USchemaDatabase::StaticClass(), FName("SchemaDatabase"), EObjectFlags::RF_Public | EObjectFlags::RF_Standalone);
+
+		int ComponentId = 100010;
```

Comment:
I would move the component starting index(100010) to spatial contstants rather than it assigned here.